### ISSUE TITLE
Escape key not cancelling dialog - edge case

### ISSUE THREAD

**Issue by ashwin0003** [quasarframework/quasar#4300]
Hi,

quasar-framework: ^0.17.19
vue: ^2.5.2

Using escape when `q-dialog` is open cancels the dialog usually.

But in this special case pressing escape isn't cancelling the dialog.
I hope I added enough data to explain the specific case.

Listing.vue

```javascript
<template>
  <div>
    <ListingTable :loading="requestInProgress"
      :users="users"
      @block="handleBlockUser"
      @unblock="handleUnblockUser"
      @delete="handleDeleteUser"
    ></ListingTable>
  </div>
</template>

<script>
import ListingTable from './ListingTable'
...

export default {
...

data () {
  return {
    requestInProgress: false,
    users: []
  },
},

...
</script>
```

ListingTable.vue

```javascript
<template>
  <div>
    <q-table :data="$props.users"
      ...
    >
      <template slot="body" slot-scope="props">
        <q-tr :props="props"
          ...
          <q-td key="actions" :props="props">
            <UserActionSelect :user="props.row" @user-action="handleUserAction"></UserActionSelect>
          </q-td>
      </template>
    </q-table>
  </div>
</template>

<script>
import UserActionsSelect from './UserActionsSelect'
...

export default {
...

props: {
  users: { Array, required: true }
},

...

methods: {
  handleUserAction (payload) {
    this.$emit(payload.action, payload.id)
  },
...
}
```

UserActionsSelect.vue
```javascript
<template>
  <div>
    <q-btn-dropdown label="Actions">
      <q-item @click.native="select($props.user.isBlocked ? 'unblock' : 'block')" v-close-overlay>
        <q-item-main><q-item-tile label>{{ $props.user.isBlocked ? 'Unblock' : 'Block' }}</q-item-tile></q-item-main>
      </q-item>
      <q-item @click.native="select('delete')" v-close-overlay>
        <q-item-main><q-item-tile label>Delete</q-item-tile></q-item-main>
      </q-item>
    </q-btn-dropdown>

    <q-dialog v-model="showConfirmation" @ok="onConfirm" @cancel="onCancel">
      <span slot="title">Confirm user {{ selectedActionShort }}</span>
      <span slot="message">Are you sure you want to {{ selectedActionShort }} the user "{{ $props.user.username }}"</span>
    </q-dialog>
  </div>
</template>

<script>
export default {
...

props: {
  user: { required: true, Object }
},

data () {
  return {
    selectedAction: 'placeholder',
    selectedActionShort: null,
    showConfirmation: false
  }
},

methods: {
  onConfirm () {
    this.$emit('user-action', { action: this.selectedAction, id: this.$props.user.id })
    this.selectedAction = 'placeholder'
    this.selectedActionShort = null
  },

  onCancel () {
    this.selectedAction = 'placeholder'
    this.selectedActionShort = null
  },

  select (action) {
    this.selectedAction = action
  }
},

watch: {
  'selectedAction' (newVal, oldVal) {
    switch (newVal) {
      case 'block':
        this.selectedActionShort = 'block'
        break
     ...

      this.showConfirmation = true
    }
  }
}
```

One of the columns in the listing-table has a btn-dropdown. Selecting something on this dropdown will open a confirmation dialog. This dialog is not getting cancelled on pressing escape.

**Comment by nothingismagick** [comment]
It seems that your custom component is missing some required things, please
compare:
https://quasar.dev/quasar-plugins/dialog#Invoking-custom-component

>

**Comment by ashwin0003** [comment]
Don't think thats the issue.

https://v0-17.quasar-framework.org/components/dialog.html#Basic-Usage-As-a-Component
https://v0-17.quasar-framework.org/components/dialog.html#QDialog-Vue-Events

I didn't add `prevent-close` or `no-esc-dismiss` props. Even checked with vue dev tools on chrome to be sure.



And also the dialog here gets cancelled on escape-key.

Note: `User.vue` and `Listing.vue` are route components. Escape key cancels dialog when directly in `User.vue`.

User.vue

```javascript
<template>
<div>
...
  <q-btn @click.prevent="selectAction('delete')" />
...
</div>
</template>

<script>
export default {
...

methods: {
  selectAction (action) {
    this.selectedAction = action
    this.showConfirmation = true
  },

  onConfirm () {
    switch (this.selectedAction) {
      case 'delete':
        this.handleDelete()
        break
    }

    this.selectedAction = null
  },

  onCancel () {
    this.selectedAction = null
  }
}
</script>
```

**Comment by nothingismagick** [comment]
Sry - I overlooked the fact that this is a 0.17 issue

>

**Comment by rstoenescu** [comment]
Hi,

Kindly request you to update to v1. Halting development for v1 and fixing + making a fix and a new release for v0.17 is far from ideal at this very moment (huge hassle).

If this is a stop-ship for you and you cannot upgrade atm, pls tell us. But pls pls be sure this cannot be overcome by upgrading. I cannot stress enough what a hassle would be to fix this for v0.17... Thank you.
